windows/utils/get_windows_system_info.yml

```diff
@@ -33,10 +33,11 @@
   block:
     - name: "Set fact of Windows guest OS edition info"
       ansible.builtin.set_fact:
-        guest_os_edition_info: "{{ guest_os_ansible_distribution | regex_search('Microsoft (Windows|Windows Server) ([0-9]+) (.*)', '\\3') }}"
+        guest_os_edition_info: >-
+          {{ guest_os_ansible_distribution | regex_search('Microsoft Windows( Server)? ([0-9]+) (R2 )?(.*)', '\4') }}
```

Comment:
regex_search or regex_replace?

Reply:
Add `first` filter to get the one in the match list. Thanks.